src/mlpack/core/kernels/cauchy_kernel.hpp

```diff
@@ -0,0 +1,104 @@
+/**
+ * @file cauchy_kernel.hpp
+ * @author Tommi Laivamaa
+ *
+ * Implementation of the Cauchy kernel (CauchyKernel),
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_KERNELS_CAUCHY_KERNEL_HPP
+#define MLPACK_CORE_KERNELS_CAUCHY_KERNEL_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/metrics/lmetric.hpp>
+#include <mlpack/core/kernels/kernel_traits.hpp>
+
+namespace mlpack {
+namespace kernel {
+
+/**
+ * The Cauchy kernel. Given two vector @f$ x @f$, @f$ y @f$, and a bandwidth
+ * @f$ \sigma @f$ (set in the constructor),
+ *
+ * @f[
+ * K(x, y) = \frac{1}{1 + (\frac{|| x - y ||}{\sigma})^2}.
+ * @f]
+ *
+ * For more details, see the following published paper:
+ *
+ * @code
+ * @inproceedings{Basak2008,
+ *   title={A least square kernel machine with box constraints},
+ *   author={Basak, Jayanta},
+ *   booktitle={Pattern Recognition, 2008. ICPR 2008. 19th International
+ *       Conference on},
+ *   pages={1--4},
+ *   year={2008},
+ *   organization={IEEE}
+ * }
+ * @endcode
+ */
+class CauchyKernel
+{
+ public:
+  /**
+   * Default constructor; sets bandwidth to 1.0.
+   */
+  CauchyKernel() : bandwidth(1.0)
+  { }
+
+  /**
+   * Construct the Cauchy kernel with a custom bandwidth.
+   */
+  CauchyKernel(double bandwidth) : bandwidth(bandwidth)
```

Comment:
We could probably combine both constructors, and use ``bandwidth = 1.0`` as the default value.